### PR TITLE
Update tasks upon updating contacts

### LINKED ISSUE #236: [PE-D] Gen command does not copy the correct email addresses to clipboard if emails were edited
Steps I took that resulted in this bug:
1. view 1
2. edit 2 e/smth@abc.com
3. gen 1

This was what was copied to my clipboard: e111111@u.nus.edu, e123456@u.nus.edu, e234567@u.nus.edu, e345678@u.nus.edu

The above is incorrect due to the email changes as seen in the picture below

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: Decaxical/ped#5

### PR DESCRIPTION
**Pull request by snss231**
fixes #236 

also some refactoring (removed `addTask` method from `ReadOnlyTaskList`, added to `Model`/`ModelManager`)

### CODE REVIEW

File: src/main/java/seedu/address/model/TaskList.java [hunk 1]
```diff
@@ -131,6 +132,11 @@ public void removePerson(Person target) {
         internalList.forEach(task -> task.removePerson(target));
     }
 
+
+    public void setPerson(Person target, Person editedPerson) {
+        internalList.forEach(task-> task.updatePerson(target, editedPerson));
+    }
+
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/model/TaskList.java:139]: Maybe can consider adding java doc for this part?

File: src/main/java/seedu/address/model/task/Task.java [hunk 2]
```diff
@@ -92,10 +92,27 @@ public void addPerson(Person person) {
         people.add(person);
     }
 
+    /**
+     * Remove a person from the list of people associated with the task.
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/model/task/Task.java:96]:
```suggestion
     * Removes a person from the list of people associated with the task.
```

File: src/main/java/seedu/address/model/task/Task.java [hunk 3]
```diff
@@ -92,10 +92,27 @@ public void addPerson(Person person) {
...
+     *
+     * @param person Person to remove
+     */
     public void removePerson(Person person) {
         people.remove(person);
     }
 
+    /**
+     * Update a person in the list of people associated with the task.
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/model/task/Task.java:105]:
```suggestion
     * Updates a person in the list of people associated with the task.
```

File: src/main/java/seedu/address/model/task/Task.java [hunk 4]
```diff
@@ -92,10 +92,27 @@ public void addPerson(Person person) {
...
+     *
+     * @param person Person to add
+     */
+    public void updatePerson(Person person, Person editedPerson) {
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/model/task/Task.java:109]: missing out one more @param editedPerson

Comment (snss231) [reply src/main/java/seedu/address/model/task/Task.java:109]: 👌  thanks for reviewing!

**Review by brian16600 — COMMENTED** [review]
LGTM. Can go ahead and merge

**Pull request merged**